tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Mean.test.cc

```diff
@@ -0,0 +1,99 @@
+  {
+    CirclePlusGen cgen;
+
+    uint32_t weight_buf = cgen.addBuffer(std::vector<float>(8 * 2, 0.f));
+    uint32_t bias_buf = cgen.addBuffer(std::vector<float>(8, 0.f));
+    uint32_t axis_buf = cgen.addBuffer(std::vector<int32_t>{1});
+    int input = cgen.addTensor({{1, 2}, circle::TensorType::TensorType_FLOAT32});
+    int weight = cgen.addTensor({{8, 2}, circle::TensorType::TensorType_FLOAT32, weight_buf});
+    int bias = cgen.addTensor({{8}, circle::TensorType::TensorType_FLOAT32, bias_buf});
+    int fc_output = cgen.addTensor({{1, 8}, circle::TensorType::TensorType_FLOAT32});
+    int axis = cgen.addTensor({{1}, circle::TensorType::TensorType_INT32, axis_buf});
+    int output = cgen.addTensor({{1, 1}, circle::TensorType::TensorType_FLOAT32});
+    cgen.addOperatorFullyConnected({{input, weight, bias}, {fc_output}});
+    bool keep_dims = true;
+    cgen.addOperatorMean({{fc_output, axis}, {output}}, keep_dims);
+    cgen.setInputsAndOutputs({input}, {output});
+
+    float learning_rate = 0.01f;
+    int32_t batch_size = 1;
+    cgen.addTrainInfo({circle::Optimizer::Optimizer_SGD, learning_rate,
+                       circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                       circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
+
+    _context = std::make_unique<GenModelTrainContext>(cgen.finish());
+    _context->addTrainCase(uniformTCD<float>({{{1, 3}}, {{2, 1}}}, // inputs
+                                             {{{5}}, {{3}}},       // expected
+                                             {13.3691f}            // loss
+                                             ));
+
+    _context->setBackends({"train"});
+    _context->setEpoch(4);
+
+    SUCCEED();
+  }
+}
+
+TEST_F(GenModelTrain, neg_NonTrainableOps_Mean_InvalidShape)
+{
+  CirclePlusGen cgen;
+  int input = cgen.addTensor({{1, 8}, circle::TensorType::TensorType_FLOAT32});
+  int axis = cgen.addTensor({{2}, circle::TensorType::TensorType_INT32});
+  // Invalid shape: input and output's ranks should be equal if keep_dims is true
+  int out = cgen.addTensor({{1, 2, 3}, circle::TensorType::TensorType_FLOAT32});
+  bool keep_dims = true;
+  cgen.addOperatorMean({{input, axis}, {out}, keep_dims}, circle::ActivationFunctionType_NONE);
+  cgen.setInputsAndOutputs({input, axis}, {out});
+
+  float learning_rate = 0.01f;
+  int32_t batch_size = 1;
+  cgen.addTrainInfo({circle::Optimizer::Optimizer_SGD, learning_rate,
+                     circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
+
+  _context = std::make_unique<GenModelTrainContext>(cgen.finish());
+  _context->setBackends({"train"});
+  _context->expectFailCompile();
+}
+
+TEST_F(GenModelTrain, neg_NonTrainableOps_Mean_InvalidType)
+{
+  CirclePlusGen cgen;
+  int input = cgen.addTensor({{1, 8}, circle::TensorType::TensorType_INT32});
```

Comment:
```suggestion
  // Invalid  type: input tensor type should be FLOAT32 
  int input = cgen.addTensor({{1, 8}, circle::TensorType::TensorType_INT32});
```

I referred here : https://github.com/Samsung/ONE/blob/be96c99f1a1e2a301aa3a6ddd3b9601ce35a6aa7/runtime/onert/backend/cpu/ops/MeanLayer.cc#L64-L73

Reply:
I think it may be redundant, but it's not bad with the comment.